types/react-star-rating-component/index.d.ts

```diff
@@ -0,0 +1,53 @@
+// Type definitions for react-star-rating-component 1.4
+// Project: https://github.com/voronianski/react-star-rating-component#readme
+// Definitions by: Cody Sand <https://github.com/marpstar>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import * as React from "react";
+
+interface StarRatingComponentProps {
+  /** name of the radio input */
+  name: string;
+
+  /** the value of the star rating to display. i.e. the number of filled stars */
+  value: number;
+
+  /** number of icons in rating, default `5` */
+  starCount?: number;
+
+  onStarClick?: (nextValue: number, prevValue: number, name: string) => void;
+
+  onStarHover?: (nextValue: number, prevValue: number, name: string) => void;
+
+  onStarHoverOut?: (nextValue: number, prevValue: number, name: string) => void;
+
+  /** render method for the full-star icon */
+  renderStarIcon?: (
+    nextValue: number,
+    prevValue: number,
+    name: string
+  ) => JSX.Element | string;
+
+  /** render method for the half-star icon */
+  renderStarIconHalf?: (
+    nextValue: number,
+    prevValue: number,
+    name: string
+  ) => JSX.Element | string;
+
+  /** color of selected icons */
+  starColor?: string;
+
+  /** color of non-selected icons */
+  emptyStarColor?: string;
+
+  /** is component available for editing, default `true` */
+  editing?: boolean;
+}
+
+declare class StarRatingComponent extends React.Component<
+  StarRatingComponentProps
+> {}
+
+export default StarRatingComponent;
```

Comment:
This needs to be `export = StarRatingComponent` instead of export default since the library itself is a function and there is no `default` property on it. In your tests you can still use `import StarRatingComponent from "react-star-rating-component";` by using `esModuleInterop` in tsconfig to be true

Reply:
will fix.